pkg/transport/cmd/generic-main.go

```diff
@@ -100,15 +102,31 @@ func GenericMain(transportImplementation transport.Transport) {
 		logger.Error(err, "failed to create dynamic k8s clientset for transport space")
 		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
 	}
+	ocmClientset, err := clusterclient.NewForConfig(transportRestConfig)
+	if err != nil {
+		logger.Error(err, "failed to create OCM clientset for transport space")
+		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
+	}
+
+	wdsKsInformerFactory := ksinformers.NewSharedInformerFactoryWithOptions(wdsClientset, defaultResyncPeriod)
+
+	ocmInformerFactory := clusterinformers.NewSharedInformerFactory(ocmClientset, defaultResyncPeriod)
 
-	factory := ksinformers.NewSharedInformerFactoryWithOptions(wdsClientset, defaultResyncPeriod)
+	inventoryPreInformer := ocmInformerFactory.Cluster().V1().ManagedClusters()
```

Comment:
I mentioned in other comment explanation about removing this, but anyway mentioning the following - 
usually an informer variable is called with the type of the object.
e.g., informer for pods is called `podInformer`, informer for binding is called `bindingInformer`, and informer for ManagedCluster should be called `managedClusterInformer` or just `clusterInformer`. those are the naming conventions in k8s. 
if you agree to the other comments about using configmap, then this comment is not relevant to this file but it will be relevant to the new thin controller I mentioned.

Reply:
There is an unfortunate collision of names here. The type defined at https://github.com/open-cluster-management-io/api/blob/v0.12.0/client/cluster/informers/externalversions/cluster/v1/managedcluster.go#L21 is called "ManagedClusterInformer" and yet has a method that is named "Informer" and returns something that is _also_ called an informer and, in particular, is an informer of ManagedCluster objects. I have tended to call the one that precedes the other a "PreInformer", to be clearer about which is which. When looking at code like the line that we are discussing here, nothing else explicitly says this.